language/go/package.go

```diff
@@ -107,9 +108,14 @@ func (pkg *goPackage) addFile(c *config.Config, er *embedResolver, info fileInfo
 		if info.isCgo {
 			return fmt.Errorf("%s: use of cgo in test not supported", info.path)
 		}
-		pkg.test.addFile(c, er, info)
+		var test *goTarget
+		if getGoConfig(c).testMode == fileTestMode || len(pkg.tests) == 0 {
+			pkg.tests = append(pkg.tests, goTarget{})
+		}
+		test = &pkg.tests[len(pkg.tests)-1]
```

Comment:
Could you explain why that is the correct index to use?

Reply:
Comment added. In per-file mode we'll always want to use the most-recently appended target. In defaultMode we always want to use the *only* target (which will also be the last).